### PR TITLE
fix worker reconnect errors, send cando after reconnection.

### PR DESCRIPTION
**Pull request by sadlil**

### CODE REVIEW

**Comment by sadlil** [comment]
@ashiquzzaman33 run-full-test.

File: worker/agent.go [hunk 1]
```diff
@@ -163,6 +163,11 @@ func (a *agent) reconnect() error {
 	a.conn = conn
 	a.rw = bufio.NewReadWriter(bufio.NewReader(a.conn),
 		bufio.NewWriter(a.conn))
+
+	// Send What this worker can do again to gearmand.
+	for k, v := range a.worker.funcs {
+		a.worker.addFunc(k, v.timeout)
+	}
 	a.grab()
 	a.worker.reRegisterFuncsForAgent(a)
```
Comment (ashiquzzaman33) [inline worker/agent.go]: Doesn't need
```
for k, v := range a.worker.funcs {
 		a.worker.addFunc(k, v.timeout)
}
```

```
a.worker.reRegisterFuncsForAgent(a)
a.grab() 
```
Should work (Reorder of those lines)

Comment (ashiquzzaman33) [inline worker/agent.go]: Doesn't need
```
for k, v := range a.worker.funcs {
 		a.worker.addFunc(k, v.timeout)
}
```

```
a.worker.reRegisterFuncsForAgent(a)
a.grab() 
```
Should work (Reorder of those lines)

Comment (ashiquzzaman33) [inline worker/agent.go]: Doesn't need
```
for k, v := range a.worker.funcs {
 		a.worker.addFunc(k, v.timeout)
}
```

```
a.worker.reRegisterFuncsForAgent(a)
a.grab() 
```
Should work (Reorder of those lines)

**Pull request merged**